static/js/muting.js

```diff
@@ -51,6 +51,72 @@ exports.set_muted_topics = function (tuples) {
     });
 };
 
+exports.find_mute_message = function (stream, topic) {
+    var found = $('#topic_muted .topic_muted').filter(function () {
+        var elt = $(this);
+        return elt.data('stream') === stream && elt.data('topic') === topic;
+    });
+    if (found.length === 0) {
+        return null;
+    } else {
+        return found;
+    }
+};
+
+exports.mute_message_topic = function () {
+    var message;
+    message = current_msg_list.selected_message();
+
+    if (message === undefined) {
+        return;
+    }
+    unread.mark_message_as_read(message);
+    if (message.type === "stream") {
+        var stream = message.stream;
+        var subject = message.subject;
+        var $muted_topic_notification = $('#unmute_muted_topic_notification');
+
+        // if topic already muted notify the user. Else notify the
+        // topic and stream muted by the user.
+        if (exports.is_topic_muted(stream, subject) === true) {
+            if ($muted_topic_notification.css('display') === 'block') {
+                $muted_topic_notification.hide();
+            }
+            muting_ui.persist_and_rerender();
+            muting_ui.notify_already_muted_topic(stream, subject);
+        } else {
+            muting.mute_topic(stream, subject);
+            muting_ui.persist_and_rerender();
+            popovers.topic_ops.mute(stream, subject);
+        }
+    } else {
+        return;
+    }
+};
+
+exports.unmute_message_topic = function () {
+    var message;
+    message = current_msg_list.selected_message();
+    if (message === undefined) {
+        return;
+    }
+    unread.mark_message_as_read(message);
+    if (message.type === "stream") {
+        var stream = message.stream;
+        var subject = message.subject;
+        var mute_message = muting.find_mute_message(stream, subject);
+        if (mute_message !== null) {
+            // dismiss the message:
+            mute_message.remove();
+            muting.hide_topic_muted_alert();
+        }
+        muting.unmute_topic(stream, subject);
+        muting_ui.persist_and_rerender();
```

Comment:
This seems to not correctly hide the alert notification if you unmute a stream from a keyboard while the notification is still present.